sql/analyzer/catalog.go

```diff
@@ -29,9 +29,10 @@ import (
 )
 
 type Catalog struct {
-	MySQLDb       *mysql_db.MySQLDb
-	InfoSchema    sql.Database
-	StatsProvider sql.StatsProvider
+	MySQLDb            *mysql_db.MySQLDb
+	InfoSchema         sql.Database
+	SystemSchemaConfig *sql.SystemSchemaConfig
```

Comment:
This seems fine for the time being, but ideally GMS wouldn't not need to treat the pg_catalog schema differently from any other database it handles. We could handle all of this on the Dolt / Doltgres side (new database provider that knows about the pg_catalog database), it's just currently difficult to plumb the necessary information through when building the engine.

But that's hard to do, and I don't have an issue with doing this as a quick fix.